iptstate.cc

```diff
@@ -223,24 +227,84 @@ unsigned int digits(unsigned long x)
 }
 
 /*
- * Check to ensure an IP is valid
+ * Check to ensure an IP & netmask are valid
  */
-bool check_ip(const char *arg, in6_addr *addr, uint8_t *family)
+bool check_ip(const char *arg, in6_addr *addr, uint8_t *family, uint8_t *netmask, bool *has_netmask)
 {
+  const char *p_arg;
+  char ip[NAMELEN];
+
+  // Check for netmask prefix
+  p_arg = strrchr(arg, '/');
+  if (p_arg == NULL) {
+    memcpy(ip, arg, NAMELEN);
+    *has_netmask = false;
+  } else {
+    size_t ip_len = strlen(arg) - strlen(p_arg);
+    memcpy(ip, arg, ip_len);
+    ip[ip_len] = '\0';
+
+    int tmp = atoi(arg + ip_len + 1);
```

Comment:
This moves the arg pointer forward to the memory location just after the slash (since chars take 1 byte), so we can validate the netmask.
Maybe better to write it like this?
```
char *arg_net = arg + ip_len + 1;
int tmp = atoi(arg_net);

Reply:
ohhhhh right, pointer math to get to the string that has the integer. I'm dumb, sorry I just totally missed that. Yes, if we call that like arg_net and then net_int or something, I think that'll be clearer.